src/core/load-fragment.c

```diff
@@ -1249,42 +1249,7 @@ int config_parse_exec_cpu_affinity(const char *unit,
         return 0;
 }
 
-int config_parse_exec_secure_bits(const char *unit,
-                                  const char *filename,
-                                  unsigned line,
-                                  const char *section,
-                                  unsigned section_line,
-                                  const char *lvalue,
-                                  int ltype,
-                                  const char *rvalue,
-                                  void *data,
-                                  void *userdata) {
-
-        ExecContext *c = data;
-        int r;
-
-        assert(filename);
-        assert(lvalue);
-        assert(rvalue);
-        assert(data);
-
-        if (isempty(rvalue)) {
-                /* An empty assignment resets the field */
-                c->secure_bits = 0;
-                return 0;
-        }
-
-        r = secure_bits_from_string(rvalue);
-        if (r < 0) {
-                log_syntax(unit, LOG_WARNING, filename, line, r,
-                           "Failed to parse secure bits, ignoring: %s", rvalue);
-                return 0;
-        }
-
-        c->secure_bits = r;
-
-        return 0;
-}
+DEFINE_CONFIG_PARSE_ENUM(config_parse_exec_secure_bits, secure_bits, int, "Failed to parse secure bits");
```

Comment:
hmm, so secure_bits_from_string() is misnamed, it returns -errno instead of -1. It should be renamed, and hence DEFINE_CONFIG_PARSE() be used here, rather than DEFINE_CONFIG_PARSE_ENUM